Allow available preset folder selection

AlwinEsch opened the pull request:
Before was 2016 files in one folder, some was before present in different folders and overwritten several times during install.

This allow to select the wanted source by the addon settings.

To make this possible, unfortunately, a patch was required within ProjectM, as this always packed all the files only in one folder.

But find it so much better, as some of them were just for testing purposes and they do not really want to see them.

Regarding Debian something should be done, because this is not compatible (and apparently not before).

![Screenshot_20191005_145202](https://user-images.githubusercontent.com/6879739/66255095-f6dc3400-e77f-11e9-8366-c3a0be47d65e.png)

AlwinEsch committed:
udpate .gitignore

AlwinEsch committed:
increase version to 2.3.1

revmischa commented:
Mischa Collection best collection 👍

revmischa commented on visualization.projectm/resources/language/resource.language.en_gb/strings.po:
```diff
@@ -74,7 +74,49 @@ msgctxt "#30013"
+msgstr ""
+
+msgctxt "#30024"
+msgid "Mischa Collection"
```
```suggestion
msgid "Mischa's Collection"
```

revmischa commented on depends/common/projectm/0001-install-presets-in-separate-dirs.patch:
```diff
@@ -0,0 +1,27 @@
+-	test -z $(DESTDIR)$(pkgdatadir) || $(MKDIR_P) $(DESTDIR)$(pm_presets_dir)
+-	find "$(PRESETSDIR)" -type f -print0 | LC_ALL=C sort -z | xargs -0 '-I{}' $(INSTALL_DATA) '{}' $(DESTDIR)$(pm_presets_dir)
++	find "$(PRESETSDIR)" -type f -exec install -Dm 755 "{}" "$(DESTDIR)/$(pm_data_dir)/{}" \;
++#	test -z $(DESTDIR)$(pkgdatadir) || $(MKDIR_P) $(DESTDIR)$(pm_presets_dir)
```
maybe we could make this a [configure flag](https://github.com/projectM-visualizer/projectm/blob/master/configure.ac)?

AlwinEsch replied:
Sounds good, I try and create a request to ProjectM itself.

AlwinEsch committed:
separate available presets to folders

Before was 2016 files in one folder, some was before present in different folders and overwritten several times during install.
This allow to select the wanted source be the addon settings.
